js/parts/RangeSelector.js

```diff
@@ -948,13 +949,29 @@ RangeSelector.prototype = {
             } else if (select) {
                 selectedExists = true; // Only one button can be selected
                 state = 2;
+                btnSelected = i;
             }
 
             // If state has changed, update the button
             if (button.state !== state) {
                 button.setState(state);
             }
         });
+
+        // When all buttons are unchecked selected property
+        // should be cleared (#9209)
+        if (
+            btnSelected === undefined &&
+            selected !== null &&
+            selected !== undefined
+        ) {
+            rangeSelector.setSelected(null);
+        } else if (
+            btnSelected !== undefined &&
+            btnSelected !== selected
+        ) {
+            rangeSelector.setSelected(btnSelected);
+        }
```

Comment:
Consider to replace the whole condition with:
```
 if (btnSelected === undefined) {
           if(H.defined(selected) {
                 rangeSelector.setSelected(null);
           } else if (btnSelected !== selected) {
                 rangeSelector.setSelected(btnSelected);
           }
}
```

Why? Because in two IFs you check the same part `btnSelected !== undefined`.